themes/default/content/registry/packages/kubernetes/how-to-guides/managing-resources-with-server-side-apply.md

```diff
@@ -0,0 +1,759 @@
+---
+title: "Managing Resources with Server-Side Apply"
+meta_desc: Learn how to manage Kubernetes resources with Server-Side Apply
+layout: how-to-guide
+---
+
+[Server-Side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) (SSA) is a resource management strategy that was introduced in Kubernetes `v1.18`. Clients using SSA can safely share the management of Kubernetes resources by making the API Server responsible for computing diffs and resolving conflicts.
+
+The [v3.20.0 release](https://github.com/pulumi/pulumi-kubernetes/releases/tag/v3.20.0) of the Pulumi Kubernetes provider added support for managing resources using SSA. This feature is currently opt-in using the `enableServerSideApply` [Provider option]({{< relref "/registry/packages/kubernetes/api-docs/provider">}}), but will become the default in the next major release. Using SSA provides the following benefits:
+
+1. Kubernetes resources may be safely managed by more than one controller.
+2. It is now possible to "Upsert" resources; create the resource if it does not exist, or apply the configuration to an existing resource.
+3. It is now possible to patch resources with the Patch resource types in the SDK. Each resource type in the SDK has a corresponding Patch resource.
+4. The `last-applied-configuration` annotation is no longer used.
+
+This guide will show you how to enable SSA support and use these features to manage Kubernetes resources.
+
+## Enable Server-Side Apply
+
+Use the `enableServerSideApply` [Provider option]({{< relref "/registry/packages/kubernetes/api-docs/provider">}}) option to enable Server-Side Apply for the Provider.
+
+{{< chooser language "typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as kubernetes from "@pulumi/kubernetes";
+
+const provider = new kubernetes.Provider("k8s", {enableServerSideApply: true});
```

Comment:
I assume I can also use config instead?

`pulumi config set kubernetes:enableServerSideApply true`

if I am not using explicit providers? Should we call this out?

Reply:
Yes, good point. I'll update to show the alternative config methods.